sdk/mediaservices/Microsoft.Azure.Media.LiveVideoAnalytics.Edge/src/Microsoft.Azure.Media.LiveVideoAnalytics.Edge.csproj

```diff
@@ -16,6 +16,7 @@
     <PackageReference Include="Microsoft.Azure.Devices.Client" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.Authentication" />
     <PackageReference Include="System.ValueTuple" />
+    <PackageReference Include="Newtonsoft.Json" VersionOverride="12.0.3"/>
```

Comment:
Why do you need to explicitly override this dependency here?

Reply:
default one is using 10.03, we need the 12 version as the devices.client needs that dependency.. I didn't want to change the global version as a lot of other projects use that.

Reply:
@AlexGhiondea What is your feeling on this? We have the global version to ensure that libraries don't become incompatible with each other. If we don't feel comfortable upgrading all our libraries then we risk a customer getting blocked because they cannot use your library with other libraries. It would be much better that we catch that here then having the customer find it and potentially get blocked.

Reply:
Azure.Devices libraries are not generated as part of the repo, they have their own repo, so the frameworks, and dependency version are not matching your versions.

Reply:
@weshaggard raises a great point. Once we change this dependency we are now in a place where we don't know if this library will work with any of the other libraries that use an older version of Newtonsoft.

Is the change to version 12 in the Azure.Devices library a new change?

Reply:
Yes, the other version we were using earlier was dependent on the older version of newtonsoft.
the stable libraries we are using and all the newer libraries, needs the 12.0.3

Reply:
I will start a different discussion with their team regarding the targetframework and other differences with this repo. and their plan to support other Azure libraries.